app/scripts/modules/core/src/application/config/trafficGuard/trafficGuardConfig.component.ts

```diff
@@ -97,11 +89,8 @@ export class TrafficGuardConfigController {
     this.config.forEach(guard => {
       this.clusterMatches.push(
       this.application.clusters.filter(c => {
-        const clusterNames = this.namingService.parseClusterName(c.name);
-        return (guard.account === '*' || guard.account === c.account) &&
-          (guard.location === '*' || c.serverGroups.map(s => s.region).includes(guard.location)) &&
-          (guard.stack === '*' || clusterNames.stack === (guard.stack || '')) &&
-          (guard.detail === '*' || clusterNames.freeFormDetails === (guard.detail || ''));
+        return c.serverGroups.map(s => s.region)
```

Comment:
Do you need this first map? Couldn't it just be:
```
      this.application.clusters.filter(c => c.serverGroups.some(s => ClusterMatcher.getMatchingRule(c.account, s.region, c.name, [guard]) !== null))
        .map(c => {
          return {
            name: c.name,
            account: guard.account,
            regions: guard.location === '*' ? uniq(c.serverGroups.map(g => g.region)).sort() : [guard.location]
          };
        })
      );
```

Reply:
ugh, yes